PolyEngine/Engine/Src/TimeSystem.cpp

```diff
@@ -0,0 +1,60 @@
+#include "EnginePCH.hpp"
+
+#include "TimeSystem.hpp"
+#include <chrono>
+
+using std::chrono::steady_clock;
+using std::chrono::duration_cast;
+using std::chrono::duration;
+
+//------------------------------------------------------------------------------
+void Poly::TimeSystem::TimeUpdatePhase(World * world)
+{
+	TimeWorldComponent* timeComponent = &world->GetTimeWorldComponent();
+
+	//Get new delta time
+	std::chrono::steady_clock::time_point frameTime = steady_clock::now();
+	duration<float> deltaTime = duration_cast<duration<float>>(frameTime - timeComponent->LastFrameTime);
```

Comment:
I agree, we should keep all calculations in timers in double. Actually it would be nice to have simple wrapper class for Time and TimePoint.